src/backend/distributed/deparser/deparse_extension_stmts.c

```diff
@@ -0,0 +1,266 @@
+#include "lib/stringinfo.h"
+#include "nodes/pg_list.h"
+#include "nodes/parsenodes.h"
+#include "utils/builtins.h"
+
+/* Local functions forward declarations for helper functions */
+static void AppendCreateExtensionStmt(StringInfo buf, CreateExtensionStmt *stmt);
+static void AppendDropExtensionStmt(StringInfo buf, DropStmt *stmt);
+static void AppendExtensionNameList(StringInfo buf, List *objects);
+static void AppendAlterExtensionSchemaStmt(StringInfo buf,
+										   AlterObjectSchemaStmt *alterExtensionSchemaStmt);
+static void AppendAlterExtensionStmt(StringInfo buf,
+									 AlterExtensionStmt *alterExtensionStmt);
+
+
+/*
+ * GetExtensionOption returns palloc'd string value of DefElem node with
+ * "defname" from "options" list
+ */
+const char *
+GetExtensionOption(List *extensionOptions, const char *defname)
+{
+	const char *targetStr = NULL;
+
+	ListCell *defElemCell = NULL;
+
+	foreach(defElemCell, extensionOptions)
+	{
+		DefElem *defElement = (DefElem *) lfirst(defElemCell);
+
+		if (IsA(defElement, DefElem) && strncmp(defElement->defname, defname,
+												NAMEDATALEN) == 0)
+		{
+			targetStr = strVal(defElement->arg);
+			break;
+		}
+	}
+
+	/* return target string safely */
+	if (targetStr)
+	{
+		return pstrdup(targetStr);
+	}
+	else
+	{
+		return NULL;
+	}
+}
+
+
+/*
+ * DeparseCreateExtensionStmt builds and returns a string representing the
+ * CreateExtensionStmt to be sent to worker nodes.
+ */
+const char *
+DeparseCreateExtensionStmt(CreateExtensionStmt *createExtensionStmt)
+{
+	StringInfoData sql = { 0 };
+	initStringInfo(&sql);
+
+	AppendCreateExtensionStmt(&sql, createExtensionStmt);
+
+	return sql.data;
+}
+
+
+/*
+ * AppendCreateExtensionStmt appends a string representing the CreateExtensionStmt to a buffer
+ */
+static void
+AppendCreateExtensionStmt(StringInfo buf, CreateExtensionStmt *createExtensionStmt)
+{
+	const char *extensionName = createExtensionStmt->extname;
+
+	List *optionsList = createExtensionStmt->options;
+
+	/*
+	 * We do only fetch "new_version" and "schema" options from optionList
+	 * as we will append "IF NOT EXISTS" and "CASCADE" clauses regardless
+	 * of statement's content before propagating it to worker nodes.
+	 * We also do not care old_version for now.
+	 */
+	const char *schemaName = GetExtensionOption(optionsList, "schema");
+	const char *newVersion = GetExtensionOption(optionsList, "new_version");
+
+	Assert(schemaName != NULL);
+
+	/*
+	 * We do not check for if schemaName is NULL as we append it in deparse
+	 * logic if it is not specified.
+	 */
+	schemaName = quote_identifier(schemaName);
+
+	appendStringInfo(buf, "CREATE EXTENSION IF NOT EXISTS %s WITH SCHEMA %s",
+					 extensionName, schemaName);
+
+	/* "new_version" may not be specified in CreateExtensionStmt */
+	if (newVersion)
+	{
+		newVersion = quote_identifier(newVersion);
+
+		appendStringInfo(buf, " VERSION %s", newVersion);
+	}
+
+	appendStringInfoString(buf, " CASCADE;");
```

Comment:
I think we should only include CASCADE if the user includes it. Can you change that?

Reply:
done at 41be1d8